website/static/js/formViewModel.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Maintains the base class for knockoutJS form ViewModels
+ */
+'use strict';
+
+var ko = require('knockout');
+
+var $osf = require('js/osfHelpers');
+var oop = require('js/oop');
+
+var ValidationError = oop.extend(Error, {
+    constructor: function (messages, header, level) {
+        this.super.constructor.call(this);
+        this.messages = messages || [];
+        this.header = header || 'Error';
+        this.level = level || 'warning';
+    }
+});
+
+var FormViewModel = oop.defclass({
+    /**
+    * Base class KO viewmodel based forms should inherit from.
+    *
```

Comment:
Ack. ecmascript is so weird.

On Tue, Mar 24, 2015 at 3:11 PM, Steven Loria <notifications@github.com>
wrote:

> In website/static/js/formViewModel.js
> <https://github.com/CenterForOpenScience/osf.io/pull/2140#discussion_r27062713>
> :
>
> > +var $osf = require('js/osfHelpers');
> > +var oop = require('js/oop');
> > +
> > +var ValidationError = oop.extend(Error, {
> > +    constructor: function (messages, header, level) {
> > +        this.super.constructor.call(this);
> > +        this.messages = messages || [];
> > +        this.header = header || 'Error';
> > +        this.level = level || 'warning';
> > +    }
> > +});
> > +
> > +var FormViewModel = oop.defclass({
> > +    /**
> > +    * Base class KO viewmodel based forms should inherit from.
> > +    *
>
> Forgot to comment here; docstrings should go above classes/functions/etc.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CenterForOpenScience/osf.io/pull/2140/files#r27062713>
> .
>



-- 
*Harry Rybacki* - Developer, Center for Open Science
Research: http://scinet.osf.io
Blog: http://rybacki.io
[image: http://github.com/harryrybacki]
<http://github.com/HarryRybacki> [image:
https://twitter.com/HarryRybacki] <https://twitter.com/HarryRybacki> [image:
http://www.linkedin.com/in/HarryRybacki/]
<http://www.linkedin.com/in/HarryRybacki/> [image:
https://plus.google.com/+HarryRybacki]
<https://plus.google.com/+HarryRybacki>